### PR TITLE
Ax 539 athlete page add market price line to apt price chart

### PR DESCRIPTION
**Pull request by pearlson16**
# Description
athlete page add market price line to apt price chart

Fixes Jira Ticket # https://athletex.atlassian.net/browse/AX-539

image

## Type of change
Please delete options that are not relevant.
- [ ] New feature (non-breaking change which adds functionality)

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have assigned 2 reviewers to check my work
- [ ] My changes generate no new warnings

### CODE REVIEW

File: lib/athlete/bloc/athlete_page_bloc.dart [hunk 1]
```diff
@@ -77,17 +82,39 @@ class AthletePageBloc extends Bloc<AthletePageEvent, AthletePageState> {
...
-          final priceRecord = await mlbRepo.getPlayerPriceHistory(playerId,);
-          updatePriceGraphData(priceRecord, emit);
+          final priceRecord = await mlbRepo.getPlayerPriceHistory(
+            playerId,
+          );
+          final recordLength = priceRecord.priceHistory.length;
+          if (recordLength > 0) {
+            final startDate = priceRecord.priceHistory[0].timestamp;
+            final marketRecords = await athleteUseCase.getMarketPriceHistory(
+              nflRepo,
```
Comment (Friendlygeckos) [inline lib/athlete/bloc/athlete_page_bloc.dart:112]: `mlbRepo`

File: lib/repositories/subgraph/sub_graph_repo.dart [hunk 2]
```diff
@@ -52,21 +52,23 @@ class SubGraphRepo {
...
+    String token, {
+    String startDate = '',
+  }) async {
     // calculating the first time of 24 hours ago as secondsSinceEpoch
-    final startTime = (DateTime.now()
-                .subtract(const Duration(days: 1))
-                .millisecondsSinceEpoch /
-            1000)
-        .round();
+    final startTime = startDate != ''
```
Comment (Friendlygeckos) [inline lib/repositories/subgraph/sub_graph_repo.dart:65]: why would there not be a `startDate`?

File: lib/athlete/widgets/build_long_chart.dart [hunk 3]
```diff
@@ -38,6 +38,32 @@ SfCartesianChart buildLongChart(
         width: 2,
         opacity: 1,
         dashArray: <double>[5, 5],
+      ),
+      FastLineSeries<GraphData, DateTime>(
+        name: 'Market Value',
+        dataSource: chartStats.toSet().toList(),
```
Comment (Friendlygeckos) [inline lib/athlete/widgets/build_long_chart.dart:44]: What is going on over there? converting to a set which removes duplicate values and then converting it back to a list. I thought this was handled somewhere else.

File: lib/athlete/bloc/athlete_page_bloc.dart [hunk 4]
```diff
@@ -98,16 +125,43 @@ class AthletePageBloc extends Bloc<AthletePageEvent, AthletePageState> {
...
+  ) {
+    var longIndex = 0, shortIndex = 0;
+    final longHistory = marketRecord.longRecord.priceHistory;
+    final shortHistory = marketRecord.shortRecord.priceHistory;
+
+    final graphStats = priceRecord.priceHistory.map(
+      (record) {
+        final bookDate = DateFormat('yyy-MM-dd').parse(record.timestamp);
+        if (longHistory.isNotEmpty) {
+          final nextIndex = min(longIndex + 1, longHistory.length - 1);
```
Comment (Friendlygeckos) [inline lib/athlete/bloc/athlete_page_bloc.dart:140]: Do you mind explaining the logic behind this?

File: lib/athlete/bloc/athlete_page_bloc.dart [hunk 5]
```diff
@@ -46,6 +50,7 @@ class AthletePageBloc extends Bloc<AthletePageEvent, AthletePageState> {
   final TokensRepository _tokensRepository;
   final MLBRepo mlbRepo;
   final NFLRepo nflRepo;
+  final GetScoutAthletesDataUseCase athleteUseCase;
```
Comment (Friendlygeckos) [inline lib/athlete/bloc/athlete_page_bloc.dart:53]: `getScoutAthletesDataUseCase`

**Review by Friendlygeckos — CHANGES_REQUESTED** [review]

File: lib/athlete/bloc/athlete_page_bloc.dart (hunk 1 above)
Comment (MichaelFrank0317) [reply lib/athlete/bloc/athlete_page_bloc.dart:112]: fixed it to mlbRepo

File: lib/repositories/subgraph/sub_graph_repo.dart (hunk 2 above)
Comment (MichaelFrank0317) [reply lib/repositories/subgraph/sub_graph_repo.dart:65]: we need to send startDate if we fetch pair data from QuestDB

File: lib/athlete/widgets/build_long_chart.dart (hunk 3 above)
Comment (MichaelFrank0317) [reply lib/athlete/widgets/build_long_chart.dart:44]: removed unnecessary converting

**Review by MichaelFrank0317 — APPROVED** [review]

**Review by Friendlygeckos — APPROVED** [review]

**Pull request merged**